classes/addons/advanced-custom-fields.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * Class BEA_CSF_Addon_Post_Types_Order
+ *
+ * Addon for Post Types Order
+ * @see : https://fr.wordpress.org/plugins/post-types-order/
+ *
+ * @author Amaury BALMER
+ */
+class BEA_CSF_Addon_ACF {
+	static $acf_fields = array();
+
+	public function __construct() {
+		if ( !function_exists('acf_get_field_groups' ) ) {
+			return false;
+		}
+
+		add_action( 'bea_csf.client.posttype.merge', array( __CLASS__, 'bea_csf_client_posttype_merge' ), 10, 3 );
+		return true;
+	}
+
+	public static function bea_csf_client_posttype_merge( $data, $sync_fields, $new_post ) {
+		$groups = acf_get_field_groups( array( 'post_type' => $new_post->post_type ) );
+		if ( empty( $groups ) ) {
+			return false;
+		}
+
+		$fields = array();
+		foreach ( $groups as $group ) {
+			$fields += acf_get_fields( $group );
+		}
+
+		// Get only fields
+		self::prepare_acf_fields( $fields );
+
+		// Reloop on meta from sync
+		if ( isset( $data['meta_data'] ) && is_array( $data['meta_data'] ) && ! empty( $data['meta_data'] ) ) {
```

Comment:
Est-ce que l'on peut éviter une imbrication avec le test inverse et un return ?

Reply:
Why not.

Reply:
J'ai fait la modification FYI.